### PR TITLE
Projects v2

### PR DESCRIPTION
**Pull request by laander**
Motivation
------------
We're working on a revised Projects model at Timekit that will be more native to the platform. With this also comes an update to our availability engine (successor to FindTime) and Microsoft related endpoints.

Added new endpoints:
```
// Availability
timekit.fetchAvailability({ ... })

// Projects
timekit.getProjects()
timekit.getProject({ id })
timekit.getHostedProject({ slug })
timekit.getEmbedProject({ id })
timekit.createProject({ ... })
timekit.updateProject({ id, ... })
timekit.deleteProject({ id })
timekit.addProjectResource({ id, ... })
timekit.setProjectResources({ id, resources })
timekit.removeProjectResource({ id, resourceId })

// Account
timekit.getAccounts()
timekit.accountGoogleSignup({ callback }, shouldAutoRedirect)
timekit.accountGoogleSync()
timekit.accountMicrosoftSignup({ callback }, shouldAutoRedirect)
timekit.accountMicrosoftSync()
```

Side-effects/other
------------
Also improved the DX of the endpoints list in README so it includes a specification of which object properties you can send in the arguments to the methods.

Tests
------------
Updated tests with new endpoints.

Release dependencies
------------
https://github.com/timekit-io/booking-js/pull/176

Who should review it
------------
@Trolzie @vistik

### CODE REVIEW

**Comment by Trolzie** [comment]
@laander should we consider adding some info about `/widgets` endpoint to let people know that this is an _old_ endpoint, and they should be using *projects* instead?

File: README.md [hunk 1]
```diff
@@ -120,83 +120,108 @@ timekit.getConfig();
...
 // Auth endpoints (Note: only used to fetch a resource key)
-timekit.auth(data);
+timekit.auth({ ... })
 
 // Account endpoints
-timekit.getAccounts();
-timekit.accountGoogleSignup(data, shouldAutoRedirect:Boolean);
-timekit.accountSync(data);
+timekit.getAccounts()
+timekit.accountGoogleSignup({ callback }, shouldAutoRedirect)
```
Comment (Trolzie) [inline README.md]: what about microsoft? does it use the same? should we consider renaming/adding?

**Comment by laander** [comment]
@Trolzie I have not mentioned in the README on purpose. Although it was public in our API reference at some point, so people don't get any funky ideas and start to use it. If there is a comment about it on http://developers.timekit.io/ , dont you think that would be enough?

File: README.md (hunk 1 above)
Comment (laander) [reply README.md]: Ah, good point, I hadn't actually considered that. Let me check in the API routes

**Comment by Trolzie** [comment]
You are right, docs should be where we do this. I just thought that both internally, plus if people inspect, then it is always nice to see a comment like:

```
// widgets endpoint is no longer encouraged
// use: ´/projects´ instead for future-proofing
```

I know its a bad example, but you know what i mean :D

**Comment by laander** [comment]
@Trolzie I like that, good idea, all add some comments in the code 👍

**Comment by laander** [comment]
@Trolzie I like that, good idea, all add some comments in the code 👍

**Comment by Trolzie** [comment]
swell @laander I'll wait for the next push 💃

**Comment by Trolzie** [comment]
lookin good 👾

**Pull request merged**